File: python/src/estruturas/repetitivas/divisao.py
""" Problema "divisão"
Escreva um algoritmo que leia dois números e imprima o resultado da divisão do primeiro pelo segundo.
Caso não for possível, mostre a mensagem “DIVISÃO IMPOSSÍVEL”
"""

MIN_NUMBER = 1
MAX_NUMBER = 10


def divisao() -> None:
    n = int(input(f"Quantos casos você vai digitar (mínimo de {MIN_NUMBER} e máximo de {MAX_NUMBER})? "))

    if not MIN_NUMBER < n < MAX_NUMBER:
        n = MIN_NUMBER

    for i in range(1, n+1):
        print(f"Caso {i} de {n}")

        numerator = int(input("Entre com o numerador: "))
        denominator = int(input("Entre com denominador: "))

        if denominator != 0:
            division = numerator / denominator
            print(f"DIVISÃO = {division:.2f}")
        else:
            print("DIVISÃO IMPOSSÍVEL")


if __name__ == "__main__":
    divisao()

File: csharp/Source/Estruturas/Repetitivas/Divisao.cs
namespace Source.Estruturas.Repetidas;

/**
 * Problema "divisao"
 * Escreva um algoritmo que leia dois números e imprima
 * o resultado da divisão do primeiro pelo segundo.
 * Caso não for possível, mostre a mensagem “DIVISAO IMPOSSIVEL”
 */
internal static class Divisao
{
    private static readonly short MAX_NUMBERS = 10;

    public static void Execute()
    {
        Console.Write($"Quantos casos você vai digitar (máximo de {MAX_NUMBERS}? ");
        int n = int.Parse(Console.ReadLine()!);

        n = n < 1 || n > MAX_NUMBERS ? 1 : n;

        for (int i = 0; i < n; i++)
        {
            Console.WriteLine($"\nCaso {i + 1} de {n}");

            Console.Write("Entre com o numerador: ");
            int numerator = int.Parse(Console.ReadLine()!);

            Console.Write("Entre com denominador: ");
            int denominator = int.Parse(Console.ReadLine()!);

            if (denominator != 0)
            {
                double division = (double)numerator / denominator;
                Console.WriteLine("DIVISÃO = {0:F2}", division);
            }
            else
            {
                Console.WriteLine("DIVISÃO IMPOSSÍVEL");
            }
        }
    }
}